Source/CarthageKit/Project.swift

```diff
@@ -866,54 +864,56 @@ private func repositoryURLForProject(project: ProjectIdentifier, preferHTTPS: Bo
 }
 
 /// Clones the given project to the global repositories folder, or fetches
-/// inside it if it has already been cloned.
+/// inside it if it has already been cloned. Optionally takes a commitish to 
+/// check for prior to fetching.
 ///
 /// Returns a signal which will send the operation type once started, and
 /// the URL to where the repository's folder will exist on disk, then complete
 /// when the operation completes.
-public func cloneOrFetchProject(project: ProjectIdentifier, preferHTTPS: Bool) -> SignalProducer<(ProjectEvent, NSURL), CarthageError> {
+public func cloneOrFetchProject(project: ProjectIdentifier, preferHTTPS: Bool, commitish: String? = nil) -> SignalProducer<(ProjectEvent, NSURL), CarthageError> {
 	let fileManager = NSFileManager.defaultManager()
 	let repositoryURL = repositoryFileURLForProject(project)
 
 	return SignalProducer.attempt { () -> Result<GitURL, CarthageError> in
-			do {
-				try fileManager.createDirectoryAtURL(CarthageDependencyRepositoriesURL, withIntermediateDirectories: true, attributes: nil)
-			} catch let error as NSError {
-				return .Failure(.WriteFailed(CarthageDependencyRepositoriesURL, error))
-			}
+		do {
+			try fileManager.createDirectoryAtURL(CarthageDependencyRepositoriesURL, withIntermediateDirectories: true, attributes: nil)
+		} catch let error as NSError {
+			return .Failure(.WriteFailed(CarthageDependencyRepositoriesURL, error))
+		}
 
-			return .Success(repositoryURLForProject(project, preferHTTPS: preferHTTPS))
+		return .Success(repositoryURLForProject(project, preferHTTPS: preferHTTPS))
 		}
```

Comment:
The code inside this block should be indented another level so that it's indented past the closing `}`.